### ISSUE TITLE
txlock should be set before new block

### ISSUE THREAD

**Issue by Alex-Werner** [dashevo/bitcore-node-dash#3]
When we send a instantSend transaction, it appears that txlock arrive later than the transaction itself.

In consequence, if we request a tx in insight-api before receiving a txlock, we set up txlock=false after seeking for a txlock in `zmqKnownTransactionLocks` (see getDetailedTransaction function (which use RPC's `getRawTransaction`) ).
Because we setup the result of the transaction in cache (valid up to next block), until a new block appear, a insight-api call to a instantSend transaction will be found as not txlock.

**Comment by gilemon** [comment]
Yes indeed, it happens if _zmqTransactionHandler is called before _zmqTransactionLockHandler.
It is not systematic.
There is setTimeout at 500ms in /Users/gilemon/tmp/nodejs/dash/insight/bitcore-node-dash/node_modules/insight-api-dash/lib/index.js to allow for receiving the zmqTransactionLockHandler. I'm testing with 700ms but anyway there is a trade-off with this solution. It's adding a delay and maybe it's still not handling 100% of the cases. I guess the best way to solve this would be to modify the ZMQ behavior on the node so that it always send txlock before tx. If that's possible?

**Comment by nmarley** [comment]
@Alex-Werner This seems pretty old now. Is this still needed?

**Comment by nmarley** [comment]
Can we handle txlock events as a cache-buster, e.g. similar to new blocks, a txlock event would clear the cache? Or just drop the cache entirely?

Personally not sure if this is a good idea -- caching at the app level. Another approach would be to handle this at the deployment level instead, and put a caching server between the Node.js app and the clients.
